src/commands/consolidate.py

```diff
@@ -0,0 +1,330 @@
+            count=count,
+        )
+
+    click.secho(
+        f'Consolidating {len(target_source_public_keys)} validators: ',
+    )
+    for target_key, source_key in target_source_public_keys:
+        click.secho(f'    {source_key} -> {target_key}')
+    if not no_confirm:
+        click.confirm(
+            'Proceed consolidation?',
+            default=True,
+            abort=True,
+        )
+    current_fee = await get_request_fee(
+        settings.network_config.CONSOLIDATION_CONTRACT_ADDRESS, block_number=chain_head.block_number
+    )
+    if current_fee > MAX_CONSOLIDATION_REQUEST_FEE:
+        logger.info(
+            'Consolidation is skipped because high consolidation fee, current fees is %s. '
+            'Increase MAX_CONSOLIDATION_REQUEST_FEE env var to allow it',
+            current_fee,
+        )
+        return
+    protocol_config = await get_protocol_config()
+
+    oracle_signatures = await poll_consolidation_signature(
+        target_source_public_keys=target_source_public_keys,
+        vault=vault_address,
+        protocol_config=protocol_config,
+    )
+
+    encoded_validators = _encode_validators(target_source_public_keys)
+    tx_hash = await submit_consolidate_validators(
+        vault_address=vault_address,
+        validators=encoded_validators,
+        oracle_signatures=oracle_signatures,
+        current_fee=current_fee,
+    )
+
+    if tx_hash:
+        click.secho(
+            'Validators has been successfully consolidated',
+            bold=True,
+            fg='green',
+        )
+
+
+async def _get_all_target_source_public_keys(
+    vault_address: ChecksumAddress, block_number: BlockNumber, count: int | None = None
+) -> list[tuple[HexStr, HexStr]]:
+    """
+    Fetch all available public keys pairs for consolidation.
+    Can be limited via count parameter.
+    """
+    logger.info('Fetching available validators for consolidation...')
+    vault_contract = VaultContract(vault_address)
+    public_keys = await vault_contract.get_registered_validators_public_keys(
+        from_block=settings.network_config.KEEPER_GENESIS_BLOCK,
+        to_block=block_number,
+    )
+    active_balances = await fetch_active_validators_balances(public_keys)
+    if not active_balances or len(active_balances) < 2:
+        raise click.ClickException(
+            f'Not enough active validators for vault {vault_address} to consolidate'
+        )
+
+    source_target_public_keys = _split_validators(active_balances)
+    if count is not None:
+        source_target_public_keys = source_target_public_keys[:count]
+    return source_target_public_keys
+
+
+async def _get_selected_target_source_public_keys(
+    from_keys: list[HexStr],
+    to_key: HexStr,
+) -> list[tuple[HexStr, HexStr]]:
+    """Validate that provided public keys can be consolidated."""
+    logger.info('Checking selected validators for consolidation...')
+    active_balances = await fetch_active_validators_balances(from_keys + [to_key])
```

Comment:
You also must check that `to_key` is 0x02 and `from_keys` are 0x01 and all of the validators are registered through the vault.